### ISSUE TITLE
where is the set_trainable method?

### ISSUE THREAD

**Issue by jiangxiluning** [fg91/visualizing-cnn-feature-maps#5]

**Comment by fg91** [comment]
The code currently requires `fastai==0.7.0` (unfortunately, I haven't found the time yet to update the code to the new fastai v1). The `env.yml` file contains the required packages, please follow the instructions in the README to create a conda environment.

**Comment by KylinA1** [comment]
Is this method necessary?
Since we already put the model into eval mode by `model.eval()`

**Comment by ssaneei** [comment]
@gf1 I tried installing the fastai 0.7.0 but there's a problem with torch<0.4.
And the same error about set_trainable(), what does it suppose to do?
@KylinA1 you mean with the same arguments?

**Comment by ssaneei** [comment]
> @gf1 I tried installing the fastai 0.7.0 but there's a problem with torch<0.4.
> And the same error about set_trainable(), what does it suppose to do?
> @KylinA1 you mean with the same arguments?

I found [this documentation](https://raw.githubusercontent.com/fastai/fastai/0fc78e8e8696464cc7eddf7b37764e1ab716402b/doc/_build/html/_modules/fastai/core.html) and "translated" it with ChatGPT, it's saying:


1. **`set_trainable_attr(m, b)`**: This function sets the `trainable` attribute of a model `m` to the boolean value `b`. Additionally, it sets the `requires_grad` attribute of all parameters in `m` to `b`. This function can be used to control whether a model's parameters should be updated during training (`b=True`) or frozen (`b=False`).

2. **`apply_leaf(m, f)`**: This function applies a given function `f` to each leaf module in the model `m`. A leaf module is defined as a module that does not contain any child modules. If `m` is an instance of `nn.Module`, the function `f` is directly applied to `m`. If `m` has child modules, the function `apply_leaf` is recursively called on each child module, applying `f` to them.

3. **`set_trainable(l, b)`**: This function uses `apply_leaf` to set the trainable attribute of all leaf modules in the model `l` to the boolean value `b`. It achieves this by calling `set_trainable_attr` with `b` as the argument for each leaf module.


_____
Let me know if it's not the case :)
